### ISSUE TITLE
Configuration cache checks fingerprint of included build against properties from root build

### ISSUE THREAD

**Issue by bamboo** [gradle/gradle#15820]
Consider a composite build with the following structure:
```
build-logic/
    build.gradle.kts
       println(providers.gradleProperty("property").forUseAtConfigurationTime().orNull)
    gradle.properties
       property=42
settings.gradle.kts
    includeBuild("build-logic")
```

### Expected Behavior

Configuration cache is reused the second time `gradle help --configuration-cache` is ran.

### Current Behavior

Configuration cache is not reused because Gradle property `property` has changed.

**Comment by autonomousapps** [comment]
I ran into this but from a slightly different angle. Consider the following
```
// main build
build.gradle
gradle.properties
buildSrc/
  build.gradle
```
Both the main build and buildSrc's build.gradle referenced a property with `providers.gradleProperty('mavenRepoUrl')`, and this property is defined in `gradle.properties`. When building with configuration cache, on the second time, I get this warning
> Calculating task graph as configuration cache cannot be reused because Gradle property 'mavenRepoUrl' has changed.

I was able to resolve this problem by symlinking gradle.properties into buildSrc.

**Comment by ZakTaccardi** [comment]
I don't have an exact reproducer, but was seeing:

```
Calculating task graph as configuration cache cannot be reused because Gradle property 'someProperty' has changed.
```

Even when `someProperty` was uniquely named and not re-used across different builds. This only occured in `includedBuild` scenarios - if I ran the code directly in the project as a non-included build, the configuration cache was correctly re-used.

Until this is fixed, I wouldn't be able to mix configuration cache with included builds

**Comment by bamboo** [comment]
This issue has been fixed in #19079 and the fix will be available in the upcoming 7.4 RC1.

There's a similar limitation related to system properties defined in an included build properties file which remains open, see #19184.
